### ISSUE TITLE
Program's weights update

### ISSUE THREAD

**Issue by Rivendile** [mfranzs/meta-learning-curiosity-algorithms#3]
Hi, it's said that **the neural network weights of the programs are updated via gradient descent** in the paper 2.2. However, in the code, the intrinsic reward calculated in **internal_rewards.py/_internal_reward** is detached. I'm quite confused about this. Could you please tell me how the weights are updated in your code?

**Comment by mfranzs** [comment]
The neural network weights in the curiosity algorithm are not trained as part of the PPO back-propagation step; they are thus detached in internal_rewards.py to prevent this. Instead, loss values are added through `AddToLoss` operations (in operations.py), collected in the `values_to_add_to_loss` array in executor.py, and optimized on line 123 in executor.py.

**Comment by Rivendile** [comment]
It solves my confusion. Thanks a lot!!!
